commit 21d1fde64f343c5c01969e0323c4b64359718b17 (refs/changes/57/957/3)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-11-29T09:05:33-08:00 (2 years, 10 months ago)
    
    TOOLS-1591 'sdcadm up' of all "stateless" zones is accidentally giving them a delegate dataset
    Reviewed by: Pedro Palazón Candel <pedro@joyent.com>
    Approved by: Pedro Palazón Candel <pedro@joyent.com>

diff --git a/CHANGES.md b/CHANGES.md
index 9ffa95a..151acd8 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -10,6 +10,13 @@
 
 # sdcadm Changelog
 
+## 1.15.0
+
+- TOOLS-1591: 'sdcadm up' of a number of core Triton services (the simple
+  "stateless" ones, e.g. vmapi, napi, papi, etc.) will ensure the instance
+  has a delegate dataset if required (if the SAPI service says it should
+  have one).
+
 ## 1.14.1
 
 - joyent/sdcadm#28 Allow 'sdcadm create portolan --skip-ha-check' to work.
diff --git a/lib/procedures/update-stateless-services-v1.js b/lib/procedures/update-stateless-services-v1.js
index 8b4ae26..4056a50 100644
--- a/lib/procedures/update-stateless-services-v1.js
+++ b/lib/procedures/update-stateless-services-v1.js
@@ -98,17 +98,24 @@ UpdateStatelessServicesV1.prototype.execute = function ussv1Execute(opts, cb) {
             ]);
         }
         if (change.inst) {
-            // Only for the Docker service instances:
-            steps.push(function (_, next) {
-                s.ensureDelegateDataset({
-                    service: change.service,
-                    progress: opts.progress,
-                    zonename: change.inst.zonename,
-                    log: opts.log,
-                    server: change.inst.server
-                }, next);
-            });
-            // Some svcs might not have an instance (e.g. the manta zone).
+            /*
+             * If the service params require a delegate dataset, then ensure
+             * the instance has one.
+             */
+            assert.optionalBool(change.service.params.delegate_dataset,
+                'change.service.params.delegate_dataset');
+            if (change.service.params.delegate_dataset) {
+                steps.push(function (_, next) {
+                    s.ensureDelegateDataset({
+                        service: change.service,
+                        progress: opts.progress,
+                        zonename: change.inst.zonename,
+                        log: opts.log,
+                        server: change.inst.server
+                    }, next);
+                });
+            }
+
             steps = steps.concat([
                 s.updateVmUserScript,
                 s.imgadmInstall,
diff --git a/package.json b/package.json
index 9019689..38164a2 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "sdcadm",
   "description": "Administer a SmartDataCenter (SDC) standup",
-  "version": "1.14.1",
+  "version": "1.15.0",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
